src/sonic-yang-models/yang-models/sonic-bgp-neighbor.yang

```diff
@@ -48,7 +48,56 @@ module sonic-bgp-neighbor {
 
     container sonic-bgp-neighbor {
         container BGP_NEIGHBOR {
+            list BGP_NEIGHBOR_TEMPLATE_LIST {
+                description "This list is to support template based BGP neighbor configuration handled by bgpcfgd";
+                key "neighbor";
+
+                leaf neighbor {
+                    type inet:ip-address;
+                    description "BGP Neighbor address";
+                }
+
+                leaf asn {
+                    type uint32 {
+                        range "1..4294967295";
+                    }
+                    description "Peer AS number";
+                }
+
+                leaf holdtime {
+                    type uint16;
+                    description "Hold time";
+                }
+
+                leaf keepalive {
+                    type uint16;
+                    description "Keepalive interval";
+                }
+
+                leaf local_addr {
+                    type inet:ip-address;
+                    description "Local source address or interface name to use for connection.";
+                }
+
+                leaf name {
+                    type string;
+                    description "Peer description";
+                }
+
+                leaf nhopself {
```

Comment:
Ideally BGP neighbor config must be on below page:
https://github.com/Azure/SONiC/wiki/Configuration

@alexrallen : would you like to add BGP neighbor config section on this page.
FYI: @lguohan

Reply:
I can bring it up to date.